src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.DefaultPipeReader.cs

```diff
@@ -34,11 +36,11 @@ public DefaultPipeReader(Pipe pipe)
 
             public override void OnWriterCompleted(Action<Exception, object> callback, object state) => _pipe.OnWriterCompleted(callback, state);
 
-            public bool IsCompleted => _pipe.IsReadAsyncCompleted;
+            public ValueTaskSourceStatus GetStatus(short token) => _pipe.GetReadAsyncStatus();
 
-            public ReadResult GetResult() => _pipe.GetReadAsyncResult();
+            public ReadResult GetResult(short token) => _pipe.GetReadAsyncResult();
 
-            public void OnCompleted(Action continuation) => _pipe.OnReadAsyncCompleted(continuation);
+            public void OnCompleted(Action<object> continuation, object state, short token, ValueTaskSourceOnCompletedFlags flags) => _pipe.OnReadAsyncCompleted(continuation, state, flags);
```

Comment:
Tests that we observe flags?

Reply:
There are tests that use the flags to make sure the behavior implemented correctly (flowing the execution context and the sync context). What other kind of observation are you looking for?

Reply:
Test that sets up current execution context and passes ValueTaskSourceOnCompletedFlags.None, checking that things works as expected.

Reply:
So more of a unit of this specific functionality? Today we test this by setting async locals and making sure they exist in the callback when it fires.